client/src/services/auth/login.ts

```diff
@@ -0,0 +1,21 @@
+import { client, setAccessToken, setRefreshToken } from '../client';
+
+interface Params {
+  username: string;
+  password: string;
+}
+
+interface Response {
```

Comment:
There will be multiple response body types with the same name `Response`, wouldn't it be confusing?

Reply:
Maybe you are right.  But I think in most case we don't need them.

My point of same naming is to make every service file has same export name of same kind of variables or types.  So that developers don't need to open the file and find it.

I think in most components, will not import more than one service file, otherwise maybe a smell of too complex. Developers can still rename them as they want when import multiple services.  

My suggestion is, if you need combine multiple services in one user action, you can create a  separate service file just in your component folder.  As this service very depends on the component's business logic.   And all the service return promise, you can connect them by `then` or `Promise.all` like a work flow.  Then import the combined service  into you component.  That will make you UI component focus on display first.

BTW, I also created type utilities in `/src/types/service.d.ts`.  You can use `ServiceResponse<T>` or `ServiceParams<T>` to extract the `Response` or `Params` type from `typeof service`.  This will be more convenient to use.

Reply:
Understood